core/src/main/java/org/apache/calcite/util/Sarg.java

```diff
@@ -170,6 +170,14 @@ public boolean isComplementedPoints() {
             .allMatch(RangeSets::isPoint);
   }
 
+  /**
+   * Returns whether this Sarg can be expanded to more simple form, e.g.
+   * the IN call or single comparison.
+   */
+  public boolean isSimple() {
```

Comment:
There is no objective definition of 'simple'. I would claim that 'x <> 5' is simple, and so is 'x IS NOT NULL'. You might disagree.

So, this method doesn't belong on `Sarg`. It belongs in whichever piece of code needs a particular definition of 'simple'.